test/views/declaration/DeclarationHolderRequiredViewSpec.scala

```diff
@@ -0,0 +1,109 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.declaration
+
+import base.Injector
+import config.AppConfig
+import controllers.declaration.routes
+import forms.common.YesNoAnswer
+import forms.common.YesNoAnswer.YesNoAnswers
+import helpers.views.declaration.CommonMessages
+import models.requests.JourneyRequest
+import models.{DeclarationType, Mode}
+import org.jsoup.nodes.Document
+import services.cache.ExportsTestData
+import unit.tools.Stubs
+import views.components.gds.Styles
+import views.declaration.spec.UnitViewSpec
+import views.html.declaration.declarationHolder.declaration_holder_required
+import views.tags.ViewTest
+
+@ViewTest
+class DeclarationHolderRequiredViewSpec extends UnitViewSpec with ExportsTestData with CommonMessages with Stubs with Injector {
+  private val form = YesNoAnswer.form()
+  private val declarationHolderRequiredPage = instanceOf[declaration_holder_required]
+  private def view(implicit request: JourneyRequest[_]): Document =
+    declarationHolderRequiredPage(Mode.Normal, form)
+
+  "Declaration Holder Required View on empty page" should {
+
+    "have correct message keys" in {
+
+      messages must haveTranslationFor("declaration.declarationHolderRequired.title")
+      messages must haveTranslationFor("declaration.declarationHolderRequired.hint.1")
+      messages must haveTranslationFor("declaration.declarationHolderRequired.hint.2")
+      messages must haveTranslationFor("declaration.declarationHolderRequired.tradeTariff.link")
+      messages must haveTranslationFor("declaration.declarationHolderRequired.help.bodyText")
+      messages must haveTranslationFor("declaration.declarationHolderRequired.empty")
+    }
+  }
+
+  "Declaration Holder Required View on empty page" should {
+
+    onEveryDeclarationJourney() { implicit request =>
+      "display page title" in {
+        view.getElementsByClass(Styles.gdsPageLegend) must containMessageForElements("declaration.declarationHolderRequired.title")
+      }
+
+      "display section header" in {
+        view.getElementById("section-header") must containMessage("declaration.section.2")
+      }
+
+      "display page hints" in {
+        view.getElementById("declaration-holder-required-hint1") must containMessage("declaration.declarationHolderRequired.hint.1")
+
+        val hint2 = view.getElementById("declaration-holder-required-hint2")
+        hint2.childNodeSize() mustBe 3
+        hint2.child(0) must haveHref(instanceOf[AppConfig].tradeTariffUrl)
```

Comment:
I'd rather create a field with `AppConfig` instead. Bear in mind you are creating the real object for every journey (5 times). With a field it would be a small optimization.

But the other thing is whether it would be better to mock `AppConfig` ? There is an `OverridableInjector` class that allows to create real objects with dependencies defined by you. Just saying it might be worth considering.

Reply:
Let's chat about mocking AppConfig. I still need to figure out what's in the `OverridableInjector` class.